### PR TITLE
conformance: Add tests for omitted and empty backendRefs

### PR DESCRIPTION
**Pull request by optix2000**
**What type of PR is this?**
/kind test
/area conformance-test

**What this PR does / why we need it**:
Add tests for null/empty list `backendRefs`
The behavior is explicitly specified, but some controllers do not conform.

**Which issue(s) this PR fixes**:
N/A

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by snorwin** [comment]
@optix2000, did you also open an issue on the implementation's repository to let them know about this problem?

File: conformance/tests/httproute-omitted-backendrefs.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+				Request:   http.Request{Path: "/forward"},
+				Backend:   confsuite.InfraBackendServiceNameV1,
+				Namespace: ns,
+			}, {
+				Request:  http.Request{Path: "/omitted-no-forward"},
+				Response: http.Response{StatusCode: 404},
+			}, {
+				Request:  http.Request{Path: "/empty-no-forward"},
+				Response: http.Response{StatusCode: 404},
+			},
```
Comment (snorwin) [inline conformance/tests/httproute-omitted-backendrefs.go:60]: An other edge case which might be worth covering, when backends exist but all have `weight: 0`. 

However, the behavior isn't clearly defined by the spec:
> If weight is set to 0, no traffic should be forwarded for this entry.

https://gateway-api.sigs.k8s.io/reference/1.5/spec/#httpbackendref

File: conformance/tests/httproute-omitted-backendrefs.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2026 The Kubernetes Authors.
```
Comment (snorwin) [inline conformance/tests/httproute-omitted-backendrefs.go]:
```suggestion
Copyright The Kubernetes Authors.
```

**Review by snorwin — CHANGES_REQUESTED** [review]

**Comment by optix2000** [comment]
> @optix2000, did you also open an issue on the implementation's repository to let them know about this problem? 

Lets just say that they don't agree it's a problem because the comformance tests pass.

File: conformance/tests/httproute-omitted-backendrefs.go (hunk 1 above)
Comment (optix2000) [reply conformance/tests/httproute-omitted-backendrefs.go:60]: I believe `httproute-weight.go` already tests for this.
https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/tests/httproute-weight.go#L69

If 0 weight backend gets a response, it should break here.
https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/utils/weight/weight.go#L114

Comment (snorwin) [reply conformance/tests/httproute-omitted-backendrefs.go:60]: I don't fully agree, the existing tests only have one backend with `weight: 0`. The edge case where all backends are set to `weight: 0` isn't covered yet.

**Comment by howardjohn** [comment]
FWIW I had a bug this exact test would have caught (a while ago) and was surprised it wasn't covered. So :+1: from me

**Comment by youngnick** [comment]
I agree, we mandate this behavior, we should test it.

/approve

File: conformance/tests/httproute-omitted-backendrefs.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	"sigs.k8s.io/gateway-api/pkg/features"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteNoBackendRefs)
+}
+
+var HTTPRouteNoBackendRefs = confsuite.ConformanceTest{
+	ShortName:   "HTTPRouteNoBackendRefs",
+	Description: "HTTPRoute rules with omitted or empty backendRefs explicitly respond with 404",
```
Comment (youngnick) [inline conformance/tests/httproute-omitted-backendrefs.go]: Note that this is actually incorrect. The HTTPRoute docs say that this must respond with a `500`, not a `404`. This is because having an invalid or nonexistent backendRef is an _infrastructure_ problem, not a _server_ one. So it needs to be a 500-class error, but the more specific ones don't match. So we had to go with `500`.

Please change all these to be `500`, not `404`.

My bad for the previous approval, I missed this on my initial read-through.

/unapprove

Comment (optix2000) [reply conformance/tests/httproute-omitted-backendrefs.go]: @youngnick Maybe I missed a different page that defines it, but the spec [here](https://gateway-api.sigs.k8s.io/api-types/httproute/#backendrefs-optional) says 404 should be returned.

> BackendRefs defines API objects where matching requests should be sent. If unspecified, the rule performs no forwarding. If unspecified and no filters are specified that would result in a response being sent, a 404 error code is returned.

File: conformance/tests/httproute-omitted-backendrefs.go (hunk 1 above)
Comment (optix2000) [reply conformance/tests/httproute-omitted-backendrefs.go:60]: I'm happy to add it, but it seems out of scope of this PR

File: conformance/tests/httproute-omitted-backendrefs.go (hunk 3 above)
Comment (optix2000) [reply conformance/tests/httproute-omitted-backendrefs.go]: Ah ok I see in the actual API docs it's different and https://github.com/kubernetes-sigs/gateway-api/pull/4758 was just opened to fix the site docs.

File: conformance/tests/httproute-omitted-backendrefs.go (hunk 1 above)
Comment (snorwin) [reply conformance/tests/httproute-omitted-backendrefs.go:60]: @youngnick what response code would be correct this case?

**Review by snorwin — APPROVED** [review]
@optix2000 thanks, lgtm.

I defer the final review to @howardjohn.

**Comment by rikatz** [comment]
/lgtm
Thanks @optix2000

**Comment by optix2000** [comment]
@snorwin @rikatz Should this test be backported given this spec has been around since forever?

**Comment by rikatz** [comment]
yes apparently this is a behavior that wasn't being verified but is common among implementations.

I will open the cherry-pick and defer to release managers (@kflynn @bexxmodd ) 

/cherry-pick release-1.5
